python_modules/dagster/dagster_tests/execution_tests/misc_execution_tests/test_async.py

```diff
@@ -33,3 +38,48 @@ async def aio_gen(_):
 
     result = wrap_op_in_graph_and_execute(aio_gen)
     assert result.output_value() == "done"
+
+
+def test_can_run_in_async():
+    @op
+    def emit():
+        return 1
+
+    async def _go():
+        result = wrap_op_in_graph_and_execute(emit)
+        assert result.output_value() == 1
+
+    asyncio.run(_go())
+
+
+def test_aio_resource():
+    class AioResource(ConfigurableResource):
+        _loop = PrivateAttr()
+
+        @property
+        def loop(self):
+            return self._loop
+
+        @asynccontextmanager
+        async def yield_for_execution(self, context):
+            await asyncio.sleep(0)
+            self._loop = asyncio.get_running_loop()
+            yield self
+
+    @op
+    async def aio_op(aio_resource: AioResource):
+        await asyncio.sleep(0)
+        assert aio_resource.loop is asyncio.get_running_loop()
+        return "done"
+
+    result = wrap_op_in_graph_and_execute(aio_op, {"aio_resource": AioResource()})
+    assert result.output_value() == "done"
+
+    with pytest.raises(
+        DagsterError, match="Unable to handle resource with async def yield_for_execution"
+    ):
+        build_op_context({"aio_resource": AioResource()})
```

Comment:
Make sure to add to build_asset_context as well? Not sure if we support async def asset